pageserver/src/tenant/timeline.rs

```diff
@@ -613,14 +613,21 @@ impl Timeline {
             Ok(()) => Ok(()),
             seqwait_error => {
                 drop(_timer);
-                let walreceiver_status = self.walreceiver.status().await;
+                let walreceiver_status = {
+                    match &*self.walreceiver.lock().unwrap() {
+                        None => "stopping or stopped".to_string(),
+                        Some(walreceiver) => match walreceiver.status() {
+                            Some(status) => status.to_human_readable_string(),
+                            None => "Not active".to_string(),
+                        },
+                    }
+                };
```

Comment:
Is there a reason to keep this out of the `Err` branch below? It's not needed otherwise?

Reply:
Aah missed that, makes sense.